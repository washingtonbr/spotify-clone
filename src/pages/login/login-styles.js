import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100vw;
  min-height: 100vh;
  background-color: ${(props) => props.theme.bgColor};
  color: ${(props) => props.theme.secondaryColor};
`;

export const Box = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Logo = styled.img`
  width: 60px;
  height: 60px;

  @media(min-width: 768px) {
    width: 64px;
    height: 64px;
  }
`;

export const Text = styled.h1`
  font-weight: 700;
  font-size: 20px;
  color: ${(props) => props.theme.primaryColor};
  margin: 55px 0;

  @media(min-width: 768px) {
    font-size: 32px;
    margin: 65px 0;
  }
`;

export const Button = styled.button`
  display: block;
  width: 220px;
  height: 40px;
  font-weight: 700;
  font-size: 12px;
  color: ${(props) => props.theme.secondaryColor};
  line-height: 40px;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  background-color: ${(props) => props.theme.greenColor};
  border-radius: 50px;
  cursor: pointer;
  border: 0;

  @media(min-width: 768px) {
    width: 260px;
    height: 50px;
    line-height: 50px;
    font-size: 14px;
  }
`;
