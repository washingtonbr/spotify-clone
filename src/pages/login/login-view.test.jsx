import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LoginView from './login-view';
import { Button } from './login-styles';

const mockRedirectToSpotify = jest.fn();

describe('LoginView', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <LoginView
          redirectToSpotify={mockRedirectToSpotify}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should redirect to Spotify when click in login button', () => {
    const wrapper = shallow(
      <LoginView
        redirectToSpotify={mockRedirectToSpotify}
      />,
    );

    wrapper
      .find(Button)
      .first()
      .simulate('click');

    expect(mockRedirectToSpotify).toHaveBeenCalledTimes(1);
  });
});
