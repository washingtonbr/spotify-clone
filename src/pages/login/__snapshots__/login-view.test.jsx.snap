// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView renders correctly 1`] = `
<div
  className="sc-AxjAm bvmcgx"
>
  <div
    className="sc-AxirZ dcYQlC"
  >
    <img
      alt="Spotify"
      className="sc-AxiKw faXdmn"
      src="test-file-stub"
    />
    <h1
      className="sc-AxhCb gRZKvI"
    >
      Pesquise, favorite e ou√ßa!
    </h1>
    <button
      className="sc-AxhUy dGQXjA"
      onClick={[MockFunction]}
    >
      Conecte-se via Spotify
    </button>
  </div>
</div>
`;
